src/test/java/com/notably/logic/commands/suggestion/SearchSuggestionCommandTest.java

```diff
@@ -0,0 +1,148 @@
+    private static AbsolutePath toRoot;
+    private static AbsolutePath toCs2103;
+    private static AbsolutePath toCs3230;
+    private static AbsolutePath toCs2103Week1;
+    private static AbsolutePath toCs2103Week2;
+    private static AbsolutePath toCs2103Week3;
+    private static AbsolutePath toCs2103Week1Lecture;
+    private static Model model;
+    private static Body toCs2103Week1LectureBody = new Body("Week 1 is ezpz. I am falling asleep.");
+    private static Body toCs2103Week2Body = new Body("I was wrong. This mod is so time consuming");
+    private static Body toCs2103Week3Body = new Body("Will I fail this mod?");
+    private static Body toCs3230Body = new Body("I confirm fail this mod. Fail lah fail lah.");
+
+    private static final String COMMAND_WORD = "search";
+    private static final String WHITE_SPACE = "        ";
+
+    @BeforeAll
+    public static void setUp() throws InvalidPathException {
+        // Set up paths
+        toRoot = AbsolutePath.fromString("/");
+        toCs2103 = AbsolutePath.fromString("/CS2103");
+        toCs3230 = AbsolutePath.fromString("/CS3230");
+        toCs2103Week1 = AbsolutePath.fromString("/CS2103/Week1");
+        toCs2103Week2 = AbsolutePath.fromString("/CS2103/Week2");
+        toCs2103Week3 = AbsolutePath.fromString("/CS2103/Week3");
+        toCs2103Week1Lecture = AbsolutePath.fromString("/CS2103/Week1/Lecture");
+
+        // Set up model
+        BlockModel blockModel = new BlockModelImpl();
+        SuggestionModel suggestionModel = new SuggestionModelImpl();
+        ViewStateModel viewStateModel = new ViewStateModelImpl();
+        model = new ModelManager(blockModel, suggestionModel, viewStateModel);
+
+        // Add test data to model
+        Block cs2103 = new BlockImpl(new Title("CS2103"));
+        Block cs3230 = new BlockImpl(new Title("CS3230"));
+        model.addBlockToCurrentPath(cs2103);
+        model.addBlockToCurrentPath(cs3230);
+
+        Block week1 = new BlockImpl(new Title("Week1"));
+        Block week2 = new BlockImpl(new Title("Week2"));
+        Block week3 = new BlockImpl(new Title("Week3"));
+        model.setCurrentlyOpenBlock(toCs2103);
+        model.addBlockToCurrentPath(week1);
+        model.addBlockToCurrentPath(week2);
+        model.addBlockToCurrentPath(week3);
+
+        Block lecture = new BlockImpl(new Title("Lecture"));
+        model.setCurrentlyOpenBlock(toCs2103Week1);
+        model.addBlockToCurrentPath(lecture);
+
+        // Add body to the blocks
+        model.setCurrentlyOpenBlock(toCs2103Week1Lecture);
+        model.updateCurrentlyOpenBlockBody(toCs2103Week1LectureBody);
+
+        model.setCurrentlyOpenBlock(toCs2103Week2);
+        model.updateCurrentlyOpenBlockBody(toCs2103Week2Body);
+
+        model.setCurrentlyOpenBlock(toCs2103Week3);
+        model.updateCurrentlyOpenBlockBody(toCs2103Week3Body);
+
+        model.setCurrentlyOpenBlock(toCs3230);
+        model.updateCurrentlyOpenBlockBody(toCs3230Body);
```

Comment:
I think it's better if you straightaway construct these blocks with their corresponding body entry. Take a look at `BlockImpl`'s  `BlockImpl(Title title, Body body)` constructor.